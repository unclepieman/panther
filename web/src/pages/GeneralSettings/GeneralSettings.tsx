/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import React from 'react';
import { Alert, Box, useSnackbar, Flex, Card, SimpleGrid, Img } from 'pouncejs';
import ErrorBoundary from 'Components/ErrorBoundary';
import PantherIcon from 'Assets/panther-plain-logo.svg';
import { pantherConfig } from 'Source/config';
import { extractErrorMessage } from 'Helpers/utils';
import CompanyInformationForm from 'Components/forms/CompanyInformationForm';
import Footer from 'Components/Footer';
import LinkButton from 'Components/buttons/LinkButton';
import withSEO from 'Hoc/withSEO';
import { useGetGeneralSettings } from './graphql/getGeneralSettings.generated';
import { useUpdateGeneralSettings } from './graphql/updateGeneralSettings.generated';
import GeneralSettingsPageSkeleton from './Skeleton';

// Parent container for the general settings section
const GeneralSettingsPage: React.FC = () => {
  const { pushSnackbar } = useSnackbar();

  const {
    loading: getGeneralSettingsLoading,
    error: getGeneralSettingsError,
    data: getGeneralSettingsData,
  } = useGetGeneralSettings();

  const [updateGeneralSettings] = useUpdateGeneralSettings({
    onCompleted: () => {
      pushSnackbar({ variant: 'success', title: `Successfully updated company information` });
    },
    onError: error => {
      pushSnackbar({
        variant: 'error',
        title:
          extractErrorMessage(error) ||
          'Failed to update company information due to an unknown error',
      });
    },
  });

  if (getGeneralSettingsLoading) {
    return <GeneralSettingsPageSkeleton />;
  }

  if (getGeneralSettingsError) {
    return (
      <Alert
        variant="error"
        title="Failed to query company information"
        description={
          extractErrorMessage(getGeneralSettingsError) ||
          'Sorry, something went wrong, please reach out to support@runpanther.io if this problem persists'
        }
      />
    );
  }

  const { displayName, email, errorReportingConsent, analyticsConsent } = getGeneralSettingsData.generalSettings; // prettier-ignore
  return (
    <>
      <Flex direction="column" minHeight="80%">
        <SimpleGrid columns={3} spacing={5}>
          <Box as="article">
            <Card px={6} py={9}>
              <ErrorBoundary>
                <CompanyInformationForm
                  initialValues={{
                    displayName,
                    email,
                    errorReportingConsent,
                    analyticsConsent,
                  }}
                  onSubmit={values => updateGeneralSettings({ variables: { input: values } })}
                />
              </ErrorBoundary>
            </Card>
          </Box>
        </SimpleGrid>
      </Flex>
      <Footer>
        <Flex spacing={170}>
          <Box>
            <Img
              src={PantherIcon}
              alt="Panther logo"
              nativeWidth={94}
              nativeHeight={20}
              mb={3}
              display="block"
            />
            <LinkButton
              external
              size="small"
              to="https://runpanther.io/pricing/"
              variantColor="navyblue"
              variant="outline"
            >
              Get Enterprise
            </LinkButton>
          </Box>
          <Flex as="dl" spacing={9} align="center">
            <Box as="section">
              <Box id="plan" as="dt" color="navyblue-100" fontSize="small" mb={1}>
                Plan
              </Box>
              <Box aria-labelledby="plan" as="dd" fontSize="medium">
                Community
              </Box>
            </Box>

            <Box as="section">
              <Box id="aws_account_id" as="dt" color="navyblue-100" fontSize="small" mb={1}>
                AWS Account ID
              </Box>
              <Box aria-labelledby="aws_account_id" as="dd" fontSize="medium">
                {pantherConfig.AWS_ACCOUNT_ID}
              </Box>
            </Box>
            <Box as="section">
              <Box id="panther_version" as="dt" color="navyblue-100" fontSize="small" mb={1}>
                Version
              </Box>
              <Box aria-labelledby="panther_version" as="dd" fontSize="medium">
                {pantherConfig.PANTHER_VERSION} ({pantherConfig.PANTHER_COMMIT})
              </Box>
            </Box>
            <Box as="section">
              <Box id="aws_region" as="dt" color="navyblue-100" fontSize="small" mb={1}>
                AWS Region
              </Box>
              <Box aria-labelledby="aws_region" as="dd" fontSize="medium">
                {pantherConfig.AWS_REGION}
              </Box>
            </Box>
          </Flex>
        </Flex>
      </Footer>
    </>
  );
};

export default withSEO({ title: 'General Settings' })(GeneralSettingsPage);
