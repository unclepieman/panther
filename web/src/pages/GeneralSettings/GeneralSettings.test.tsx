/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import React from 'react';
import { render, buildGeneralSettings, waitFor } from 'test-utils';
import GeneralSettings from './GeneralSettings';
import { mockGetGeneralSettings } from './graphql/getGeneralSettings.generated';

describe('GeneralSettings', () => {
  test('it renders the general settings page along with the footer', async () => {
    const settings = buildGeneralSettings({
      displayName: 'Panther labs',
      email: 'test@runpanther.io',
    });
    const mocks = [mockGetGeneralSettings({ data: { generalSettings: settings } })];

    const { getByText, container, getByLabelText } = render(<GeneralSettings />, {
      mocks,
    });
    await waitFor(() => expect(getByText('Company Information')).toBeInTheDocument());

    expect(getByLabelText('Company Name')).toHaveValue('Panther labs');
    expect(getByLabelText('Email')).toHaveValue('test@runpanther.io');
    expect(getByText('Save')).toBeInTheDocument();

    expect(getByText('Preferences')).toBeInTheDocument();
    expect(getByText('Report Web Application Errors')).toBeInTheDocument();
    expect(getByText('Send Product Analytics')).toBeInTheDocument();

    expect(getByText('Plan')).toBeInTheDocument();
    expect(getByText('AWS Account ID')).toBeInTheDocument();
    expect(getByText('Version')).toBeInTheDocument();
    expect(getByText('AWS Region')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
