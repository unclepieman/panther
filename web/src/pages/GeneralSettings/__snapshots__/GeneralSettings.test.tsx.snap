// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralSettings it renders the general settings page along with the footer 1`] = `
.panther-39 {
  display: grid;
  grid-gap: 20px;
  grid-template-columns: repeat(3, 1fr);
}

.panther-40 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 80%;
}



.panther-37 {
  background-color: #23344e;
  border-radius: 4px;
  padding-left: 24px;
  padding-right: 24px;
  padding-top: 36px;
  padding-bottom: 36px;
}

.panther-12 {
  margin-bottom: 24px;
}

.panther-0 {
  font-weight: 400;
  font-size: 1.125rem;
  margin-bottom: 24px;
}

.panther-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.panther-11>*:not(:last-child) {
  margin-bottom: 16px;
}

.panther-4 {
  min-height: 47px;
  position: relative;
  border: 1px solid;
  -webkit-transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  background-color: transparent;
  border-radius: 4px;
  border-color: #3e516b;
}

.panther-4:hover {
  border-color: #0081c5;
}

.panther-4:focus-within {
  border-color: #0081c5;
}

.panther-4:focus-within label {
  font-weight: 500;
  -webkit-transform: translate(6px, 4px) scale(0.65);
  -moz-transform: translate(6px, 4px) scale(0.65);
  -ms-transform: translate(6px, 4px) scale(0.65);
  transform: translate(6px, 4px) scale(0.65);
}

.panther-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.panther-1 {
  vertical-align: top;
  width: 100%;
  height: 100%;
  padding-left: 16px;
  padding-top: 20px;
  padding-bottom: 8px;
  position: relative;
  color: #f6f6f6;
  font-size: 0.875rem;
  font-weight: 500;
  background-color: transparent;
  border: 0;
  z-index: 2;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.panther-1::-webkit-input-placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  font-weight: 500;
}

.panther-1::-moz-placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  font-weight: 500;
}

.panther-1:-ms-input-placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  font-weight: 500;
}

.panther-1::placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  font-weight: 500;
}



.panther-1:focus::-webkit-input-placeholder {
  opacity: 0.4;
}

.panther-1:focus::-moz-placeholder {
  opacity: 0.4;
}

.panther-1:focus:-ms-input-placeholder {
  opacity: 0.4;
}

.panther-1:focus::placeholder {
  opacity: 0.4;
}

.panther-1:-webkit-autofill {
  -webkit-box-shadow: 0 0 0 30px #192231 inset;
  -webkit-text-fill-color: #f6f6f6;
  border-radius: 4px;
}

.panther-2 {
  pointer-events: none;
  font-size: 0.875rem;
  padding-left: 16px;
  padding-right: 16px;
  color: #bdbdbd;
  top: 0;
  position: absolute;
  transform-origin: center left;
  -webkit-transform: translate(6px, 4px) scale(0.65);
  -moz-transform: translate(6px, 4px) scale(0.65);
  -ms-transform: translate(6px, 4px) scale(0.65);
  transform: translate(6px, 4px) scale(0.65);
  -webkit-transition: -webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  font-weight: 500;
}

.panther-13 {
  font-weight: 400;
  font-size: 1.125rem;
  margin-bottom: 16px;
}

.panther-22 {
  margin-bottom: 16px;
}

.panther-19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  font-size: 0.875rem;
}

.panther-14 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  margin-right: 8px;
}

.panther-18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  border-radius: 40px;
  width: 54px;
  height: 27px;
  padding: 4px;
  -webkit-transition: background-color 0.15s linear;
  transition: background-color 0.15s linear;
  background-color: #4f4f4f;
}

.panther-15 {
  width: 21px;
  height: 21px;
  border-radius: 99999px;
  background-color: #ffffff;
  -webkit-transform: translate3D(0, 0, 0);
  -moz-transform: translate3D(0, 0, 0);
  -ms-transform: translate3D(0, 0, 0);
  transform: translate3D(0, 0, 0);
  -webkit-transition: -webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.panther-16 {
  font-weight: 700;
  font-size: 0.625rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  margin-left: 7px;
  margin-right: 7px;
  position: absolute;
  right: 0;
}

.panther-17 {
  cursor: pointer;
  position: absolute;
  opacity: 0;
}

.panther-21 {
  font-size: 0.75rem;
  color: #bdbdbd;
}

.panther-27 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  border-radius: 40px;
  width: 54px;
  height: 27px;
  padding: 4px;
  -webkit-transition: background-color 0.15s linear;
  transition: background-color 0.15s linear;
  background-color: #0081c5;
}

.panther-24 {
  width: 21px;
  height: 21px;
  border-radius: 99999px;
  background-color: #ffffff;
  -webkit-transform: translate3D(120%, 0, 0);
  -moz-transform: translate3D(120%, 0, 0);
  -ms-transform: translate3D(120%, 0, 0);
  transform: translate3D(120%, 0, 0);
  -webkit-transition: -webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.panther-25 {
  font-weight: 700;
  font-size: 0.625rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  margin-left: 7px;
  margin-right: 7px;
  position: absolute;
  left: 0;
}

.panther-30 {
  font-size: 0.75rem;
  color: #bdbdbd;
  margin-top: 0;
}

.panther-33 {
  font-size: 0.75rem;
  color: #bdbdbd;
  margin-top: 16px;
}

.panther-32 {
  -webkit-transition: color 0.1s ease-out;
  transition: color 0.1s ease-out;
  font-weight: 500;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #7fd3ff;
  margin-left: 4px;
}

.panther-32:hover {
  color: #ace2ff;
}

.panther-32:focus {
  color: #ace2ff;
}

.panther-32:active,
.panther-32[data-active=true] {
  color: #11a5f3;
}

.panther-36 {
  width: 100%;
  outline: none;
  padding-left: 20px;
  padding-right: 20px;
  height: 46px;
  font-size: 0.938rem;
  -webkit-transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  border: 1px solid;
  border-color: #0081c5;
  background-color: #0081c5;
  cursor: pointer;
  color: #f6f6f6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.panther-36:hover {
  background-color: #0a8bcf;
  border-color: #0a8bcf;
}

.panther-36:focus {
  background-color: #0a8bcf;
  border-color: #55d6ff;
}

.panther-36:active,
.panther-36[data-active=true] {
  background-color: #0077bb;
  border-color: #0077bb;
}

.panther-60 {
  background-color: #1d2a3f;
}

.panther-59 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 1214px;
  margin-left: auto;
  margin-right: auto;
  min-height: 100%;
  padding-top: 24px;
  padding-bottom: 24px;
}

.panther-58 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.panther-58>*:not(:last-child) {
  margin-right: 170px;
}

.panther-41 {
  max-width: 100%;
  margin-bottom: 12px;
  display: block;
}



.panther-43>span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.panther-42 {
  width: auto;
  outline: none;
  padding-left: 20px;
  padding-right: 20px;
  height: 24px;
  font-size: 0.75rem;
  -webkit-transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border-radius: 2px;
  border: 1px solid;
  border-color: #3e516b;
  background-color: transparent;
  cursor: pointer;
  color: #f6f6f6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.panther-42:hover {
  background-color: #465973;
  border-color: #465973;
}

.panther-42:focus {
  background-color: #465973;
  border-color: #93a6c0;
}

.panther-42:active,
.panther-42[data-active=true] {
  background-color: #253852;
  border-color: #253852;
}

.panther-57 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.panther-57>*:not(:last-child) {
  margin-right: 36px;
}

.panther-45 {
  color: #86a3c3;
  font-size: 0.75rem;
  margin-bottom: 4px;
}

.panther-46 {
  font-size: 0.875rem;
}

<div>
  <header
    id="main-header"
  />
  <div
    class="panther-40"
  >
    <div
      class="panther-39"
    >
      <article
        class="panther-5"
      >
        <div
          class="panther-37"
        >
          <form
            action="#"
          >
            <section
              class="panther-12"
            >
              <h2
                class="panther-0"
              >
                Company Information
              </h2>
              <div
                class="panther-11"
              >
                <div
                  class="panther-5"
                >
                  <div
                    class="panther-4"
                  >
                    <div
                      class="panther-3"
                    >
                      <input
                        aria-invalid="false"
                        aria-required="true"
                        class="panther-1"
                        id="displayName"
                        name="displayName"
                        placeholder="The name of the company"
                        required=""
                        type="text"
                        value="Panther labs"
                      />
                      <label
                        class="panther-2"
                        for="displayName"
                      >
                        Company Name
                      </label>
                    </div>
                  </div>
                </div>
                <div
                  class="panther-5"
                >
                  <div
                    class="panther-4"
                  >
                    <div
                      class="panther-3"
                    >
                      <input
                        aria-invalid="false"
                        aria-required="true"
                        class="panther-1"
                        id="email"
                        name="email"
                        placeholder="The company's email"
                        required=""
                        type="text"
                        value="test@runpanther.io"
                      />
                      <label
                        class="panther-2"
                        for="email"
                      >
                        Email
                      </label>
                    </div>
                  </div>
                </div>
              </div>
            </section>
            <section
              class="panther-12"
            >
              <h2
                class="panther-13"
              >
                Preferences
              </h2>
              <fieldset
                class="panther-5"
              >
                <div
                  class="panther-22"
                >
                  <div
                    class="panther-5"
                  >
                    <label
                      class="panther-19"
                    >
                      <span
                        class="panther-14"
                      >
                        Report Web Application Errors
                      </span>
                      <div
                        class="panther-18"
                      >
                        <div
                          class="panther-15"
                        />
                        <div
                          class="panther-16"
                        >
                          OFF
                        </div>
                        <input
                          aria-checked="false"
                          aria-invalid="false"
                          class="panther-17"
                          name="errorReportingConsent"
                          type="checkbox"
                          value="false"
                        />
                      </div>
                    </label>
                  </div>
                  <p
                    class="panther-21"
                    id="error-reporting-section-helper-text"
                  >
                    Send anonymized runtime exception reports 
                    <br />
                     to improve Panther
                    '
                    s reliability.
                  </p>
                </div>
                <div
                  class="panther-22"
                >
                  <div
                    class="panther-5"
                  >
                    <label
                      class="panther-19"
                    >
                      <span
                        class="panther-14"
                      >
                        Send Product Analytics
                      </span>
                      <div
                        class="panther-27"
                      >
                        <div
                          class="panther-24"
                        />
                        <div
                          class="panther-25"
                        >
                          ON
                        </div>
                        <input
                          aria-checked="true"
                          aria-invalid="false"
                          checked=""
                          class="panther-17"
                          name="analyticsConsent"
                          type="checkbox"
                          value="true"
                        />
                      </div>
                    </label>
                  </div>
                  <p
                    class="panther-30"
                    id="product-analytics-section-helper-text"
                  >
                    Send anonymized product analytics to help us 
                    <br />
                    improve Panther.
                  </p>
                </div>
                <p
                  class="panther-33"
                  id="analytics-consent-link-helper-text"
                >
                  You can read more info about our security privacy
                  <a
                    class="panther-32"
                    href="https://docs.runpanther.io/v/release-1.2/user-guide/help/security-privacy#privacy"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    here
                  </a>
                  .
                </p>
              </fieldset>
            </section>
            <button
              class="panther-36"
              disabled=""
              type="submit"
            >
              Save
            </button>
          </form>
        </div>
      </article>
    </div>
  </div>
  <footer
    id="footer"
  >
    <div
      class="panther-60"
    >
      <div
        class="panther-59"
      >
        <div
          class="panther-58"
        >
          <div
            class="panther-5"
          >
            <img
              alt="Panther logo"
              class="panther-41"
              height="20"
              src="test-file-stub"
              width="94"
            />
            <a
              class="panther-43"
              href="https://runpanther.io/pricing/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span
                class="panther-42"
                type="button"
              >
                Get Enterprise
              </span>
            </a>
          </div>
          <dl
            class="panther-57"
          >
            <section
              class="panther-5"
            >
              <dt
                class="panther-45"
                id="plan"
              >
                Plan
              </dt>
              <dd
                aria-labelledby="plan"
                class="panther-46"
              >
                Community
              </dd>
            </section>
            <section
              class="panther-5"
            >
              <dt
                class="panther-45"
                id="aws_account_id"
              >
                AWS Account ID
              </dt>
              <dd
                aria-labelledby="aws_account_id"
                class="panther-46"
              >
                111111111111
              </dd>
            </section>
            <section
              class="panther-5"
            >
              <dt
                class="panther-45"
                id="panther_version"
              >
                Version
              </dt>
              <dd
                aria-labelledby="panther_version"
                class="panther-46"
              >
                1.2.3
                 (
                12345678
                )
              </dd>
            </section>
            <section
              class="panther-5"
            >
              <dt
                class="panther-45"
                id="aws_region"
              >
                AWS Region
              </dt>
              <dd
                aria-labelledby="aws_region"
                class="panther-46"
              >
                us-west-2
              </dd>
            </section>
          </dl>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
